subcommands/version/updates.go

```diff
@@ -0,0 +1,145 @@
+package version
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"runtime"
+
+	"github.com/hashicorp/go-version"
+	tuf "github.com/theupdateframework/go-tuf/client"
+	"github.com/theupdateframework/go-tuf/data"
+)
+
+type FioctlUpdateFinder struct {
+	tuf        *tuf.Client
+	localStore *jsonFilesStore
+	platform   string
+}
+
+type FioctlUpdate struct {
+	Version *version.Version
+	Uri     string
+	Sha512  data.HexBytes
+	len     int64
+}
+
+func NewFioctlUpdateFinder(localJsonPath string) (*FioctlUpdateFinder, error) {
+	remote, err := tuf.HTTPRemoteStore(RootUrl, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+	local := &jsonFilesStore{localJsonPath}
+	tuf := tuf.NewClient(local, remote)
+
+	platform := runtime.GOOS + "/" + runtime.GOARCH
+	return &FioctlUpdateFinder{tuf, local, platform}, nil
+}
+
+type fioctlCustom struct {
+	Platform string `json:"platform"`
+	Uri      string `json:"uri"`
+	Version  string `json:"version"`
+}
+
+func (f *FioctlUpdateFinder) FindLatest() (*FioctlUpdate, error) {
+	latestVer, err := version.NewVersion(Commit)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse binary's version: %s", err)
+	}
+
+	if !f.localStore.initialized() {
+		fmt.Println("Initializing local TUF storage")
+		if err := f.tuf.Init(RootContent); err != nil {
+			return nil, err
+		}
+	}
+	if _, err = f.tuf.Update(); err != nil {
+		return nil, err
+	}
+	targets, err := f.tuf.Targets()
+	if err != nil {
+		return nil, err
+	}
+
+	var update *FioctlUpdate
+	for name, target := range targets {
+		var custom fioctlCustom
+		if err := json.Unmarshal(*target.Custom, &custom); err != nil {
+			return nil, fmt.Errorf("unable to parse TUF data for %s: %s", name, err)
+		}
+		if target.Length == 0 {
+			return nil, fmt.Errorf("target %s has invalid length: %d", name, target.Length)
+		}
+		if custom.Platform == f.platform {
+			v, err := version.NewVersion(custom.Version)
+			if err != nil {
+				return nil, fmt.Errorf("unable to parse TUF data version for %s: %s", name, err)
+			}
+			if v.Core().GreaterThan(latestVer.Core()) {
+				update = &FioctlUpdate{
+					Sha512:  target.Hashes["sha512"],
+					Version: v,
+					Uri:     custom.Uri,
+					len:     target.Length,
+				}
+				latestVer = v
+			}
+		}
+	}
+	return update, nil
```

Comment:
looks like `update` can be `nil` `if custom.Platform != f.platform` or not true at #80?

Reply:
update can be nil. this is okay. cmd.go can handle it now that i added  e6264857fa3ea20984bfbd6064aefd6721fd84bf for the update path.